New CCP Data Connector for AWS S3 Server Access Logs

Alekhya0824 opened the pull request:
Required items, please complete

Change(s):

New Data Connector Addition for AWS S3 Server Access Logs
Reason for Change(s):

New Data Connector Addition for AWS S3 Server Access Logs
Version Updated:

Yes
Testing Completed:

Yes
Checked that the validations are passing and have addressed any issues that are present:

Yes

getting single logs not duplicate 
<img width="929" height="416" alt="image" src="https://github.com/user-attachments/assets/1a28e354-c28b-4aba-b2a5-36c690344a66" />
for duplicate log testing
![Uploading image.png…]()

Alekhya0824 committed:
Aws server access logs

rahul0216 commented on Solutions/AWS_AccessLogs/Data Connectors/AwsS3ServerAccessLogsDefinition_CCP/AWSS3ServerAccessLogs_PollingConfig.json:
```diff
@@ -0,0 +1,34 @@
+[
+	{
+		"name": "AwsS3 Access Logs Polling Config",
+		"apiVersion": "2022-12-01-preview",
```
Use the stable version of API.

Alekhya0824 replied:
updated

rahul0216 commented on Solutions/AWS_AccessLogs/Data Connectors/AwsS3ServerAccessLogsDefinition_CCP/AWSS3ServerAccessLogs_DCR.json:
```diff
@@ -0,0 +1,27 @@
+{
+    "name": "AWSS3AccessLogs",
+    "apiVersion": "2021-09-01-preview",
```
Use the stable version of API.

Alekhya0824 replied:
updated

rahul0216 commented on Solutions/AWS_AccessLogs/Data Connectors/AwsS3ServerAccessLogsDefinition_CCP/AWSS3ServerAccessLogs_ConnectorDefinition.json:
```diff
@@ -0,0 +1,181 @@
+{
+    "name": "AwsS3ServerAccessLogsDefinition",
+    "apiVersion": "2022-09-01-preview",
```
Use the stable version of API.

Alekhya0824 replied:
updated

rahul0216 reviewed (commented):
Please address the inline comments.

Alekhya0824 committed:
updated the changes

Alekhya0824 commented on Solutions/AWS_AccessLogs/CloudFormationTemplates/readme.md:
```diff
@@ -0,0 +1,47 @@
+   - Role ARN: Enter the Role ARN obtained from the CloudFormation stack.
+   - Queue URL: Enter the SQS Queue URL obtained from the CloudFormation stack.
+
+7. Click "Connect" to enable the AWS Security Hub Findings connector.
```
updated

Alekhya0824 commented on Solutions/AWS_AccessLogs/CloudFormationTemplates/readme.md:
```diff
@@ -0,0 +1,47 @@
+   - Role ARN: Enter the Role ARN obtained from the CloudFormation stack.
+   - Queue URL: Enter the SQS Queue URL obtained from the CloudFormation stack.
+
+7. Click "Connect" to enable the AWS Security Hub Findings connector.
```
updated

v-maheshbh committed:
solution Packaged

v-prasadboke committed:
Refactor dataCollectionEndpointId handling in templates

Updated AWSS3ServerAccessLogs_DCR.json to use a template variable for dataCollectionEndpointId. Refactored mainTemplate.json to define and use dataCollectionEndpointId1 variable, improving maintainability and consistency. Updated 3.0.0.zip package to reflect these changes.

v-prasadboke committed:
Update ReleaseNotes.md
